### ISSUE TITLE
Is there any new windows builds available

### ISSUE THREAD

**Issue by nickbe** [garglk/garglk#285]
Since I've read about lots of bug fixes I wonder if there's any recent windows builds available somewhere 😁

**Comment by fireton** [comment]
@cspiegel 
I second that! PLEASE, PLEASE, PLEASE make new Windows release!

**Comment by fireton** [comment]
I've tried to build it myself. Tried to follow instructions in wiki. No luck.
I've tried lots of MinGW distros, no luck at all. I do not understand errors in code it gives (different every time).

The cleanest log was with distro from https://nuwen.net/mingw.html but it says that 64 bit target is not compatible with 32 dll provided. How to change target I didn't understand eigher. I'm not GNU C++ developer so I'm in dismay.

So ANYONE with enough skills PLEASE build a win32 gargoyle distro. And please consider to build it once a year at least. Once at month is better.

**Comment by nickbe** [comment]
Same thing here. I've tried to build the project a while ago, but got stuck on lots of compiler errors. There were probably simply some libs missing, but I gave up due to lack of time then.

I wonder if there's any development going on at all for some time now.

**Comment by cspiegel** [comment]
I can't speak to building on Windows, but on Linux with MinGW (and probably/possibly any other system with MinGW):

```
jam -sC++=i686-w64-mingw32-g++ -sCC=i686-w64-mingw32-gcc -sOS=MINGW -sMINGWARCH=i686-w64-mingw32 -sCROSS=1
```

This should build everything.

I've not done Windows development, nor do I have any experience with Windows installers, but I did manage to bundle everything up into a package that seems to work fine with Wine on Linux.  I have no idea what versions of Windows (if any) it will run on, how upgrades work, and so on.  If anybody wants to test it: [gargoyle-2017.11-testing-windows.zip](https://github.com/garglk/garglk/files/1438941/gargoyle-2017.11-testing-windows.zip)

At this point nobody with Windows knowledge has offered to do the builds, so it may fall to me to get it done.  For that to happen, I'd really like some feedback from Windows users on this installer.  If it seems to work well enough then I'd be willing to do the official Windows builds.

**Comment by fireton** [comment]
Many thanks, Chris! I've tested it on Windows 7 and it installed and works just fine! I'm interested in Gargoyle as z-machine games interpreter that can write cyrillic transcript. With last released version it is not so. But new Bocfel writes transcripts just fine.

Do you think some more throughout testing is needed? With other interpreters? Looks like if the program installs and runs and not crashing it will run the same on all platforms. But if you need some testing please tell, I'll try to comply.

But personally I think this version will suit for me and for other people who waiting for new Windows release. Please tag new version! :)

**Comment by curiousdannii** [comment]
> I'm interested in Gargoyle as z-machine games interpreter that can write cyrillic transcript. With last released version it is not so. But new Bocfel writes transcripts just fine.

Does Lectrote work?

@cspiegel In a few weeks time I'd be able to put some time aside for doing Windows builds. I think we need a new release across all platforms though.

**Comment by fireton** [comment]
> Does Lectrote work?

How to check it? All *.z? files I run managed by Bocfel. Sorry I'm not much into IF game formats save Inform and TADS2 since they are only supported in Russia.

**Comment by fireton** [comment]
TADS2 games do not run with this version. Reports about missing libstdc++-6.dll for tadsr.exe.

**Comment by fireton** [comment]
here: https://stackoverflow.com/questions/6404636/libstdc-6-dll-not-found/6404637
I found something that may be a solution:

> If you are using MingW to compile C++ code on Windows, you may like to add the options `-static-libgcc` and `-static-libstdc++` to link the C and C++ standard libraries statically and thus remove the need to carry around any separate copies of those. Version management of libraries is a pain in Windows, so I've found this approach the quickest and cleanest solution to creating Windows binaries.

Hope it helps.

**Comment by cspiegel** [comment]
Thanks all for the updates.  I built with a static libgcc and libstdc++, but there was still a requirement for libwinpthread as well, which I'd have to copy manually.  Since I'm doing manual copies, I'm just going to include libgcc and libstdc++ as well as libwinpthread.

I'm attaching an updated installer which includes libstdc++, so TADS and Geas should work.

Extensive testing of all interpreters would be nice, but I'm primarily interested in whether things work on Windows *at all* (so learning TADS/Geas were broken was exactly what I'm looking for).

New version: [gargoyle-2017.11-testing-windows.zip](https://github.com/garglk/garglk/files/1441348/gargoyle-2017.11-testing-windows.zip)

Windows builds were the primary blocker for a new release, so if we can agree that one of the versions I produce here is acceptable, I'm almost ready to move forward on release.  Issue #284 is a blocker but I'll just cut a new Bocfel release on demand for that one, and I'd like to finish #272, which is easy enough but I've been stalling on the cleanest way to address it.  An imminent release will be enough of an impetus for me to get these done.

**Comment by fireton** [comment]
Checked new version on Windows 10 64bit. All terps are starting fine, no errors. I think this version is good enough for release.

**Comment by nickbe** [comment]
Can you check the alan3 plugin? I think there's a newer version (beta5) available here:
https://www.alanif.se/download-alan-v3/interpreters/interpreters-3-0beta5/gargoyle-alan3-3_0beta5-win32x86zip?format=raw

**Comment by nickbe** [comment]
Also I just found there are now fonts which seem to be installed. As I remember this was not necessary before and all fonts were handled "internally" by gargoyle... or so I thought...

This made the gargoyle folder completely portable so far and there was no need to install anything at all. It could be easily run from a usb stick.
Did that change?

**Comment by cspiegel** [comment]
> Can you check the alan3 plugin?

I looked into this a little while ago and it seems that the Alan interpreter's source code is not available.  Without source, newer versions cannot be included with Gargoyle.

> Also I just found there are now fonts which seem to be installed. As I remember this was not necessary before and all fonts were handled "internally" by gargoyle... or so I thought...

The testing versions I've uploaded should include the bundled fonts.  But you're right, it looks like the older installer didn't include any fonts at all.  I've built a newer installer without the external fonts, which hopefully is closer to the 2011.1 release.  It's available here: [gargoyle-2017.11-testing-windows.zip](https://github.com/garglk/garglk/files/1443506/gargoyle-2017.11-testing-windows.zip)

It looks like the uninstaller doesn't uninstall fonts, either, so you'll have to manually remove the ones that did get installed.  They are LuxiMono, CharterBT, LiberationMono, and LibLibertine.

**Comment by nickbe** [comment]
The guy from alan provides the plugin directly

**Comment by curiousdannii** [comment]
This is an open source project and it would be irresponsible to distribute binaries from others which we don't have the source to.

I emailed Thomas from Alan about the source code but haven't had a reply yet.

**Comment by cspiegel** [comment]
> The guy from alan provides the plugin directly

Yes, but in compiled form only.  Even if we were to include third-party binaries (which I'm not inclined to do), the provided ones were certainly built against 2011.1.  I don't think the libgarglk ABI has changed, but I'm not prepared to guarantee that.  This means that the Alan interpreter would have to be built against the current master on github.  It also means that only the platforms supported by the Alan developer would have the latest version.  Other users would be left out in the cold.  On top of it all, I don't see a license for the provided binaries, which means we can't include them anyway.

Basically, if source code isn't available for an interpreter, we just can't include it.

However, I would be willing to start versioning the libgarglk shared library and include necessary headers so that third parties can more easily make use of Gargoyle's Glk implementation.

**Comment by nickbe** [comment]
Hi sent the Alan authors current mail adress to curiousdannii per mail.
:)

**Comment by thoni56** [comment]
The Alan source is available on [Bitbucket](https://bitbucket.org/alanif/alan) under Artistic license 2.0.

**Comment by curiousdannii** [comment]
Fantastic, thanks Thomas!

**Comment by nickbe** [comment]
@thoni56 - Thanks for joining us 😄

**Comment by cspiegel** [comment]
@thoni56 Excellent news, thank you so much for the pointer to the repository.  Could I trouble you to add a tag at least for the latest release?  As a guess I'd say commit 084acb1, but I'd like to be sure we're including the same version as the official release.

**Comment by thoni56** [comment]
Aah! Looks like my release procedure doesn't push the tags as it should. I'll have a look at that as a part of preparing for the next beta release (3.0beta6).

You can go ahead with HEAD if you want to try something out. I have my builds setup so that I can easily build a new slot-in against Gargoyle, so there are no source tweaks necessary as of now.

**Comment by cspiegel** [comment]
I've created a branch (#294) to upgrade Alan3 to the latest master.  I'm also including a Windows build that drops in as a replacement for the interpreter included with the latest testing builds.  If you installed the 2017.11 testing version, you should be able to drop the attached alan3.exe into C:\Program Files (x86)\Gargoyle, replacing the existing entry.

Any testing done would be helpful.

[alan3.zip](https://github.com/garglk/garglk/files/1448636/alan3.zip)

**Comment by fireton** [comment]
Tested on Windows 7 64 bit, put new Alan3 build. Seems like working fine. I even downloaded one of Alan games (never did it before :) - it playing ok.

**Comment by nickbe** [comment]
Looks good. All alan3 games seem to work. Also the debugging seems to works like expected.
Thanks.

**Comment by subv242** [comment]
I've just compiled a new Windows build from latest source with MinGW 5.1.0. Enjoy.

http://www21.zippyshare.com/v/SA5plat2/file.html

**Comment by nickbe** [comment]
Thanks mate :)

**Comment by subv242** [comment]
Hello again. Today I made some experiments with optimization. Here's a new Windows build of Gargoyle, built from the source with MinGW 7.2.0. It uses the SSE2 vector instruction set, loop unrolling, etc. It should run even faster than Windows Git. Please test.

http://www47.zippyshare.com/v/PJvC8hpj/file.html

**Comment by subv242** [comment]
**nickbe** - this build includes embedded fonts Luxi Mono and CharterBT, anyway. No need to install.

**Comment by nickbe** [comment]
Ah ok. Thanks :)

**Comment by Oreolek** [comment]
Excuse me, but this is still an issue. And all the Zippyshare links in this thread are already expired. Can someone build and release this in any way?

**Comment by subv242** [comment]
I've reuploaded the latest Windows build, enjoy.

https://yadi.sk/d/PSUb-GLs3ZUE2e

https://www94.zippyshare.com/v/j8K1xpaY/file.html

**Comment by subv242** [comment]
That's strange. No viruses on zippyshare for me.

Anyway, uploaded to Google Drive instead.

**Comment by nickbe** [comment]
Thanks a lot.

**Comment by rossato** [comment]
Any chance we can get a new release for *all* platforms?  I've been giving some thought to reviving the effort to package Gargoyle in Fedora, and the 2011.1 release requires a couple patches while the current git builds pleasantly.  It would be helpful to get a new official source tarball.

**Comment by DumpsterDoofus** [comment]
For anyone reading this, **PLEASE DO NOT DOWNLOAD OR RUN THE EXECUTABLES PROVIDED ABOVE BY @subv242**. VirusTotal shows `gargoyle.exe` is bundled with `Trojan.Winlock`, which is a common Trojan in Russia.

**Comment by thoni56** [comment]
I would suggest an actual GitHub release with uploaded artifacts.

**Comment by subv242** [comment]
> @subv242 For your download on Google Drive, VirusTotal shows `gargoyle.exe` is bundled with `Trojan.Winlock`, which is a common Trojan in Russia. Are you sure your executables are safe to run?
> 
> See https://www.virustotal.com/#/file/65da7122081b9001227b3423e8c0afb49e53579646341f97bf8e006880694c8b/detection for reference.

False positive, of course. Don't ever use DrWeb, it's quite unsafe and gives a lot of false detections. As you see, none of any other AV software shows the trojan.

**Comment by nickbe** [comment]
Can't we simply have a proper windows release here in the Github Release section? That's what it's actually made for, right?

**Comment by jwhang627** [comment]
> 
> 
> > @subv242 For your download on Google Drive, VirusTotal shows `gargoyle.exe` is bundled with `Trojan.Winlock`, which is a common Trojan in Russia. Are you sure your executables are safe to run?
> > See https://www.virustotal.com/#/file/65da7122081b9001227b3423e8c0afb49e53579646341f97bf8e006880694c8b/detection for reference.
> 
> False positive, of course. Don't ever use DrWeb, it's quite unsafe and gives a lot of false detections. As you see, none of any other AV software shows the trojan.
> 
> And what do you mean by "which is common in Russia". Seriously, I don't understand you. Being an IT specialist for more than 20 years, I simply cannot allow such malware in my public builds.

It's more than just DrWeb. The one I downloaded contains malwares detected by SecureAge APEX, McAfee, Bkav, MaxSecure, and VBA32 as well.

https://www.virustotal.com/gui/file/bac6fcf7c7dc30bf9025af0c6751e23a1c67fc2788626aad2412c6c4fa4819eb/detection

**Comment by thoni56** [comment]
Since there actually is a "new" (newer than when this issue was raised at least) Windows build, perhaps you can close it?

**Comment by rossato** [comment]
The 2019.1.1 source release has global variable issues in alan2, alan3, jacl, and gargoyle itself that prevent it from compiling.  These issues are fixed in git master, so a new release would be still helpful for creating packages for Linux distributions.  I'm happy to create a new issue for that if this issue is otherwise ready to close.

**Comment by thoni56** [comment]
If you are talking about the globla variable errors that only prevents compiling with GCC v10 (no earlier version have complain about them, although they are kind of valid), Alan 3 commit https://github.com/alan-if/alan/commit/143acbd7cf28bd45eaed02162863cde88bb819c3 contains fixes for that.

**Comment by cspiegel** [comment]
A new release will come out soon, and should build properly on most platforms.

**Comment by nickbe** [comment]
Thanks for letting us know Chris 😀
